tests/test_parsers.py

```diff
@@ -62,41 +54,86 @@ def test_gelfparser_parse_gzipped_file(fs, gelf_logger):
-    events = list(parser.parse(gelf_logger.handlers[0].stream.name))
+    with open(gelf_logger.handlers[0].stream.name) as stream:
+        events = list(parser.parse(stream))
+
     assert len(events) == 2
     assert events[0] == '{"username": "foo"}'
     assert events[1] == '{"username": "bar"}'
 
 
-# pylint: disable=invalid-name,unused-argument
-def test_gelfparser_parse_with_various_chunksizes(fs, gelf_logger):
-    """Test the GELFParser parsing using different chunksizes."""
-
-    gelf_logger.info('{"username": "foo"}')
-    gelf_logger.info('{"username": "bar"}')
-    gelf_logger.info('{"username": "baz"}')
-    gelf_logger.info('{"username": "lol"}')
+def test_gelfparser_parse_partially_invalid_file(caplog):
+    """Test the GELFParser with a file containing invalid JSON strings"""
+
+    with StringIO() as file:
+        file.writelines(
+            [
+                # This is invalid gelf but we assume it's valid in our case
+                '{"short_message": "This seems valid."}\n',
+                # Invalid json
+                "{ This is not valid json and raises json.decoder.JSONDecodeError\n",
+                # Valid json but invalid gelf - raises KeyError: key "short_message" not found
+                "{}\n",
+                # As above but raises TypeError: list indices must be integers or slices, not str
+                "[]\n",
+                # Another assumed valid gelf
+                '{"short_message": {"username": "This seems valid too."}}\n',
+            ]
+        )
+
+        file.seek(0)
+        parser = GELFParser()
+        events = list(parser.parse(file))
 
-    parser = GELFParser()
-    events = list(parser.parse(gelf_logger.handlers[0].stream.name, chunksize=1))
-    assert len(events) == 4
-    assert events[0] == '{"username": "foo"}'
-    assert events[1] == '{"username": "bar"}'
-    assert events[2] == '{"username": "baz"}'
-    assert events[3] == '{"username": "lol"}'
-
-    parser = GELFParser()
-    events = list(parser.parse(gelf_logger.handlers[0].stream.name, chunksize=2))
-    assert len(events) == 4
-    assert events[0] == '{"username": "foo"}'
-    assert events[1] == '{"username": "bar"}'
-    assert events[2] == '{"username": "baz"}'
-    assert events[3] == '{"username": "lol"}'
-
-    parser = GELFParser()
-    events = list(parser.parse(gelf_logger.handlers[0].stream.name, chunksize=10))
-    assert len(events) == 4
-    assert events[0] == '{"username": "foo"}'
-    assert events[1] == '{"username": "bar"}'
-    assert events[2] == '{"username": "baz"}'
-    assert events[3] == '{"username": "lol"}'
+    assert len(events) == 2
+    assert events[0] == "This seems valid."
+    assert events[1] == {"username": "This seems valid too."}
+
+    caplog.clear()
+    with StringIO() as file:
+        file.write("{ This is not valid json and raises json.decoder.JSONDecodeError\n")
+        file.seek(0)
+        parser = GELFParser()
+        events = list(parser.parse(file))
```

Comment:
If you also want to test the DEBUG log messages:

```suggestion
        with caplog.at_level(logging.DEBUG):
            events = list(parser.parse(file))
```